internal/pkg/vm/internal/vmcontext/invocation_context.go

```diff
@@ -408,7 +409,7 @@ func (r returnWrapper) Into(o specsruntime.CBORUnmarshaler) error {
 	// TODO: if inner is also a specsruntime.CBORUnmarshaler, overwrite o with inner.
 	b := bytes.Buffer{}
 	if r.inner == nil {
-		return fmt.Errorf("failed to unmarshal nil return")
+		r.inner = adt.Empty
```

Comment:
without this we get an error here: https://github.com/filecoin-project/specs-actors/blob/master/actors/puppet/puppet.go#L92

Reply:
Thanks. I would like to slightly change how we address this - I will add a commit to this PR.